workflows/common/csv_runner.py

```
@@ -0,0 +1,187 @@
+            if i < total_rows - 1:
+                logger.info("Waiting 5 seconds before next candidate...")
+                await asyncio.sleep(5)
+            
+            # Progress update
+            processed = i + 1
+            logger.info(f"Progress: {processed}/{total_rows} candidates processed ({processed/total_rows*100:.1f}%)")
+    
+    return results
+
+
+def save_results(results: List[Dict], output_path: str):
```

Pylint (3.3.7)

[refactor] Either all return statements in a function should return an expression, or none of them should.

(R1710)